akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSubscriber.scala

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl.io
+
+import akka.actor.{ ActorLogging, ActorRef, Deploy, Props }
+import akka.stream.actor.{ ActorSubscriberMessage, RequestStrategy }
+import akka.util.ByteString
+
+import scala.concurrent.Promise
+
+/** INTERNAL API */
+object InputStreamSubscriber {
+  def props(completionPromise: Promise[Long], bufSize: Int): Props = {
+    require(bufSize > 0, "buffer size must be > 0")
+    Props(classOf[InputStreamSubscriber], completionPromise, bufSize).withDeploy(Deploy.local)
```

Comment:
@hepin1989, I got your point. Still I'm looking for reasoning why InputStreamSubscriber is different than all other Subscrbers/Publishers actors. Cause in this case I'll change to "new.." behavior within this ticket. However, if there is only a general rule - then we need a separate ticket to make it consistent everywhere